File: packages/cactus-plugin-odap-hermes/src/test/typescript/fabric-contracts/lock-asset/chaincode-typescript/src/asset.ts
/*
  SPDX-License-Identifier: Apache-2.0
*/

import { Object, Property } from "fabric-contract-api";

@Object()
export class Asset {
  @Property()
  public docType?: string;

  @Property()
  public ID: string;

  @Property()
  public IsLock: boolean;

  @Property()
  public Size: number;
}

File: weaver/samples/fabric/go-cli/cmd/asset.go
/*
 * Copyright IBM Corp. All Rights Reserved.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package cmd

import (
	"github.com/spf13/cobra"
)

// assetExchangeCmd represents the asset command
var assetExchangeCmd = &cobra.Command{
	Use:   "asset",
	Short: "operate on an asset: exchange-all|exchange-step",
	Long: `Command does nothing by itself
operate on an asset: exchange-all|exchange-step

Example:
  fabric-cli asset exchange-all`,
	Run: func(cmd *cobra.Command, args []string) {},
}

func init() {
	rootCmd.AddCommand(assetExchangeCmd)
}
